Port code to DSL2 template

nandan75 opened the pull request:
Hi @georgiesamaha 

- Please use the script [PBS_gadi_runPipeline.sh](https://github.com/Sydney-Informatics-Hub/Somatic-shortV-nf/blob/review_testRun_from_Georgie/PBS_gadi_runPipeline.sh) for testing.
- I will push the changes with emit later.
-  I am working on the 'creating intervals' - So for now please use the fixed intervals as in the [PBS_gadi_runPipeline.sh](https://github.com/Sydney-Informatics-Hub/Somatic-shortV-nf/blob/review_testRun_from_Georgie/PBS_gadi_runPipeline.sh)


Thanks,

Nandan

nandan75 committed:
Started productionizing

nandan75 committed:
First commit ptoductionize

nandan75 committed:
First pass productionize - start of gitrepo testing on gadi

Commit:
For first review of a working pipeline

nandan75 committed:
Delete results directory

nandan75 committed:
Delete work directory

nandan75 committed:
Delete null directory

nandan75 committed:
Delete previous_logs directory

nandan75 committed:
Update README.md

nandan75 committed:
Update nextflow.config

Removed some redundant entries for outDir

Commit:
Added .gitignore and Removed multiple junk files

Commit:
Tested pipeline, updated Readme.md, get reviewd

georgiesamaha commented on PBS_gadi_runPipeline.sh:
```diff
@@ -0,0 +1,53 @@
+module load singularity
+module load gatk/4.1.8.1
+#module load gatk/4.1.4.0
+
```
Why are these modules needed (apart from singularity and nextflow)? Aren't we running gatk container? Why are there multiple versions?

nandan75 replied:
I have now removed the modules for gatk , multiple versions for nextflow
Checked the pipeline 
- Downloading the gatk singularity image on the fly (the first time)
- Using the image from the cache from next time onwards

nandan75 committed:
Tested with pon_vcf in mutect2.nf

The pipeline can now tested with a pon.vcf.gz file

georgiesamaha commented on modules/Reference/Readme.txt:
Why is this in modules?

nandan75 replied:
Removed

georgiesamaha commented on modules/scatter_files/100M_primary_interval.list:
Why are these interval lists in modules? Additionally where does the a,b,c,etc naming convention come from and what does it indicate?

nandan75 replied:
I had made the following scatter lists
- B. **GATK-style .list or .intervals** from [Intervals and interval lists ](https://gatk.broadinstitute.org/hc/en-us/articles/360035531852-Intervals-and-interval-lists)
I have now switched to the  
- A. **Picard-style .interval_list** at [Intervals and interval lists ](https://gatk.broadinstitute.org/hc/en-us/articles/360035531852-Intervals-and-interval-lists) which are available at `/g/data/er01/SIH-HPC-WGS/Reference/BQSR_apply_intervals_13`
- The interval lists from modules are now removed

georgiesamaha replied:
How come you switched?

nandan75 replied:
What do you mean? 

These intervals are in our standard `/g/data/er01/SIH-HPC-WGS/Reference/BQSR_apply_intervals_13`
Do you not want me to use them?

nandan75 replied:
- I am working with creating the interval files based on user inpu. For testing, please use the default intervals as in the [PBS_gadi_runPipeline.sh](https://github.com/Sydney-Informatics-Hub/Somatic-shortV-nf/blob/review_testRun_from_Georgie/PBS_gadi_runPipeline.sh)

georgiesamaha commented on scripts/pullContainers.e103389053:
Your logs should be git ignored.

georgiesamaha commented on scripts/pullContainers.o103389053:
Same as previous comment.

nandan75 replied:
Removed the logs and added them to .gitignore

georgiesamaha commented on nextflow.config:
```diff
@@ -1,85 +1,145 @@
+
+
+  // Intervals for 'Scatter-Gather'
+  intervalList          = ['a','b','c','d','e','f','g','h','i','j','k','l','m','n']
```
As per my comment above on scatter files. Why have you used abc instead of names that correspond to the interval? Consider how this might be useful to someone else running your workflow. If the a,b,c convention comes from somewhere and has meaning, could you explain this in a comment and add a reference.

nandan75 replied:
Same response as above 

I had made the following scatter lists

B. GATK-style .list or .intervals from [Intervals and interval lists](https://gatk.broadinstitute.org/hc/en-us/articles/360035531852-Intervals-and-interval-lists)
I have now switched to the
A. Picard-style .interval_list at [Intervals and interval lists ](https://gatk.broadinstitute.org/hc/en-us/articles/360035531852-Intervals-and-interval-lists)which are available at /g/data/er01/SIH-HPC-WGS/Reference/BQSR_apply_intervals_13
The interval lists from modules are now removed

georgiesamaha commented on nextflow.config:
```diff
@@ -1,85 +1,145 @@
+  whoami            	= false
+}
+
+params.common_biallelic_path = params.ref+'/gatk-best-practices/somatic-hg38/'+'small_exac_common_3.hg38.vcf.gz'
```
why is this params sitting outside the params scope?

nandan75 replied:
This param is not required to be set here. Removed now.

georgiesamaha commented on images/Somatic_variant_calling.png:
Can replace this file with the other image_SIH you have uploaded.

georgiesamaha commented on config/gadi.config_OLD:
If this is unused and only something you're using for your own testing or notes, dont need to push to the repo, can keep in your dev environment.

georgiesamaha commented on config/gadi.config:
```diff
@@ -1,38 +1,93 @@
 singularity {
     enabled = true
-    cacheDir = "/scratch/${params.pbs_account}/${params.whoami}/.nextflow/singularity"
+    cacheDir = "/scratch/er01/ndes8648/pipeline_work/nextflow/INFRA-83-Somatic-ShortV/singularity_cache"
```
Please revert this to something reproducible by others and refrain from hardcoded paths.

nandan75 replied:
Changed to /scratch/${params.gadi_account}/${params.whoami}/singularity

georgiesamaha commented on config/gadi.config:
```diff
@@ -1,38 +1,93 @@
-    cacheDir = "/scratch/${params.pbs_account}/${params.whoami}/.nextflow/singularity"
+    cacheDir = "/scratch/er01/ndes8648/pipeline_work/nextflow/INFRA-83-Somatic-ShortV/singularity_cache"
     autoMounts = true
-    temp = "scratch/${params.pbs_account}/${params.whoami}/containers"
```
Will likely still need to set temp directory.

nandan75 replied:
Set 
temp = "scratch/${params.gadi_account}/${params.whoami}/containers"

georgiesamaha commented on config/gadi.config:
```diff
@@ -1,38 +1,93 @@
+
+
+withName: 'mutect2' {
+        executor = 'pbspro'
```
Remember to follow the [DRY principles](https://www.baeldung.com/cs/dry-software-design-principle) when writing code. If all processes are run as pbs jobs, can set executor at the same level as `withName`, inside `process{}`, rather than for each individual process.

georgiesamaha replied:
This is still unresolved and needs to be addressed.

georgiesamaha commented on config/gadi.config:
```diff
@@ -1,38 +1,93 @@
+        memory = '20.GB'
+}
+
+withName: 'GatherVcfs_step' {
```
How come this process is named `GatherVCFs_step` rather than just `GatherVCFs`?

nandan75 replied:
Changed to  GatherVCFs now

georgiesamaha commented on config/gadi.config:
```diff
@@ -1,38 +1,93 @@
+        memory = '20.GB'
+}
+
+withName: 'getFilteredVariants_and_annotate' {
```
How come this process has a combination of captialised first letters and underscores. Keep you naming conventions consistent throughout your codebase. Makes for easy reading by yourself and others.

nandan75 replied:
I have now made the names consistent. This particular file is not used . So deleted

georgiesamaha commented on modules/process2.nf:
This can be deleted

nandan75 replied:
Deleted

georgiesamaha commented on modules/process1.nf:
This can be deleted

nandan75 replied:
Deleted

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,245 @@
+
+There are three main steps in the process of calling Somatic Short Variants:
+
+(1) **Creation of Somatic short variants Panel of Normals (PoN)** : 
+<br>This involves converting the Normal BAMs to PON.   
+The PON's are -
+  * Made from normal samples i.e. the samples derived from healthy tissue and 
+  * Their main purpose is to capture recurrent technical artifacts in order to improve the results of the variant calling analysis.
+  * Please use the github repository [SomaticShortV_makePON-nf](https://github.com/Sydney-Informatics-Hub/Somatic-shortV-makePON-nf) to create the PoN. 
+<br>
+
+(2) **Call the somatic short variants for a Tumor-Normal pair** :
+<br>The two main steps involved in calling  Somatic Short Variants are - 
+  * Generate a large set of candidate somatic variants.
+  * Filter the candidate somatic variants into a more confident set of somatic variant calls.
+
+- Calling candidate somatic short variants involves the use of the tool [Mutect2](https://gatk.broadinstitute.org/hc/en-us/articles/360035531132) which calls both SNVs and indels simultaneously by generating a local assembly of haplotypes in an active region, de-novo. When Mutect2 sees a region with somatic variations, it dis-regards the existing mapping information completely and re-assembles th ...
+
+* This is followed by calculating contamination using the tools [GetPileupSummaries](https://gatk.broadinstitute.org/hc/en-us/articles/9570416554907-GetPileupSummaries) and [CalculateContamination](https://gatk.broadinstitute.org/hc/en-us/articles/9570322332315-CalculateContamination)
+* The next step is to learn the parameters of a model for orientation bias using the tool [LearnReadOrientationModel](https://gatk.broadinstitute.org/hc/en-us/articles/360051305331-LearnReadOrientationModel)
+* The [FilterMutectCalls](https://gatk.broadinstitute.org/hc/en-us/articles/360036856831-FilterMutectCalls) tool then applies filters to the raw output of Mutect2. 
+
+
+(3) **Annotating the variants** 
+<br> An external genomic variant annotations and functional effect prediction tool [SnpEff](http://pcingola.github.io/SnpEff/) is used for annotating the filtered variants (such as amino-acid changes etc). Please refer to the above link for SnpEff details.
+
```
The formatting you've used here isn't easy to read and duplicates what is already described in the GATK best practice documentation you've linked. I suggest removing all this text or at least significantly reducing to a couple of sentences.

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,245 @@
+  - [Acknowledgements/citations/credits](#acknowledgementscitationscredits)
+
+## Description
+Somatic-shortV-nf is a Nextflow pipeline for identifying somatic short variant events in Illumina short read whole genome sequence data. 
+We have followed the [GATK Best practices](https://gatk.broadinstitute.org/hc/en-us/articles/360035894731-Somatic-short-variant-discovery-SNVs-Indels-) .  
```
Use the description already provided in the [SomaticShortV](https://github.com/Sydney-Informatics-Hub/Somatic-ShortV#somatic-shortv) README. Just remove the text about being optimised for Gadi and replace with a comment about it being optimised and configured for Gadi, Setonix, Nimbus with links to each infrastructure's webpage.

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,245 @@
+## Diagram
+
+<p align="center"> 
+<img src="./images/Somatic_variant_calling_SIH_modified.png" width="60%">
```
Can you change the name of this file please to `Somatic_variant_calling.png`

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,245 @@
+* Corresponding BAM index files  
+* Input sample sheet 
+
+This pipeline processes paired BAM files and is capable of processing multiple samples in parallel. BAM files are expected to be coordinate sorted and indexed (see [Fastq-to-BAM](https://github.com/Sydney-Informatics-Hub/Fastq-to-BAM) for an example of a best practice workflow that can generate these files).  
```
Might also want to link to nf-core/sarek

nandan75 replied:
Added  link to nf-core/sarek

georgiesamaha replied:
Where?

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,245 @@
+
+This pipeline processes paired BAM files and is capable of processing multiple samples in parallel. BAM files are expected to be coordinate sorted and indexed (see [Fastq-to-BAM](https://github.com/Sydney-Informatics-Hub/Fastq-to-BAM) for an example of a best practice workflow that can generate these files).  
+
+You will need to create a sample sheet with information about the samples you are processing, before running the pipeline. This file must be **tab-separated** and contain a header and one row per sample. Columns should correspond to sampleID, BAM-N file-path, BAM-T file-path: 
```
While we've previously used tsv files, I think using csv format instead would make this step less error prone.

nandan75 replied:
Done. The pipeline now reads from a csv and was tested.

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,245 @@
+├── images/
+├── main.nf
+├── modules/
+├── PBS_gadi_runPipeline.sh
```
this is missing the `scripts/` directory. Should also move PBS execution script to this directory when ready.

nandan75 replied:
Added scripts/

nandan75 replied:
I will move the PBS execution script to the script directory after testing.

georgiesamaha commented on README.md:
````diff
@@ -1,15 +1,245 @@
+The most basic run command for this pipeline is: 
+
+```
+nextflow run main.nf --ref reference.fasta
````
Where is your input flag?

nandan75 replied:
added

georgiesamaha commented on README.md:
````diff
@@ -1,15 +1,245 @@
+nextflow run main.nf --input sample.tsv --ref /path/to/ref --gadi-account <account> --whoami <username> -profile gadi
+```
+
+Please be aware that as of October 2023, NCI Gadi HPC queues do not have external network access. This means you will not be able to pull the workflow code base or containers if you submit your nextflow run command as a job on any of the standard job queues. NCI currently recommends you run your Nextflow head job either in a GNU screen or tmux session from the login node or submit it as a job to  ...
````
Update the date.

nandan75 replied:
Should we update the entire para:

Run the main.nf on copyq (this will download all the containers - first time) and then assign normal queues to individual resources?

georgiesamaha replied:
You have a gadi script, you can just tell users to run that.

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,245 @@
+
+Please be aware that as of October 2023, NCI Gadi HPC queues do not have external network access. This means you will not be able to pull the workflow code base or containers if you submit your nextflow run command as a job on any of the standard job queues. NCI currently recommends you run your Nextflow head job either in a GNU screen or tmux session from the login node or submit it as a job to  ...
+
+The NCI Gadi config currently runs all tasks apart from the rehead processes on the normal queue. This config uses the `--gadi-account` flag to assign a project code to all task job submissions for billing purposes. The version of Nextflow installed on Gadi has been modified to make it easier to specify resource options for jobs submitted to the cluster. See NCI's [Gadi user guide](https://opus.n ...
```
The mention of the rehead process is taken from the GSV pipeline verbatim and is not relevant here.

nandan75 replied:
Done

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,245 @@
+## Workflow summaries
+### Metadata
+
+|metadata field     | GermlineStructuralV-nf / v1.0     |
```
Not the pipeline name

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,245 @@
+
+To run this pipeline you must have Nextflow and Singularity installed on your machine. All other tools are run using containers. 
+
+|Tool         | Version  |
+|-------------|:---------|
+|Nextflow     |>=20.07.1 |
+|Singularity  |          |
+|SnpEff       |     |
```
Missing GATK and versions of other tools used

nandan75 replied:
Added

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,245 @@
+|Nextflow     |>=20.07.1 |
+|Singularity  |          |
+|SnpEff       |     |
+|R            |          |
```
Same as above.

nandan75 replied:
Added

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,245 @@
+- [Nextflow documentation](https://www.nextflow.io/docs/latest/index.html)
 
 ## Help / FAQ / Troubleshooting
+* It is essential that the reference genome you're using contains the same chromosomes, contigs, and scaffolds as the BAM files. This is [mandated by Manta](https://github.com/Illumina/manta/issues/92), which will throw an error if the BAM and FASTA files do not match. To confirm what contigs are included in your indexed BAM file, you can use Samtools idxstats:
```
Taken verbatim from GSV, remove.

nandan75 replied:
Removed

georgiesamaha commented on config/gadi.config:
I suspect this config isn't being applied to the workflow execution as the final process failed to run due to memory limits. See comment below for error.

georgiesamaha replied:
When running on the login node, I get the following error:

```
Error executing process > 'getFilteredVariants (getFilteredVariants Patient2)'

Caused by:
  Process requirement exceed available memory -- req: 20 GB; avail: 4 GB
```

nandan75 replied:
Hi @georgiesamaha 

- I have now tested the pipeline - running it with the command -  `bash PBS_gadi_runPipeline.sh`
   - Multiple  times now and confirmed. All runs well for me and using the normalq using the assigned resources! Used qtata to check
   - The last step getFilteredVariants also runs fine.
   - Ching-Yu has run it as well,  on his side.
- I am assuming It was possibly a transient issue - I am facing problems with gadi today when it goes very slow at times - qstat seems to be not working sometime ...
- **So once I push all the changes** , can you please re-run the pipeline to check.

georgiesamaha replied:
This is working fine now.

georgiesamaha commented on main.nf:
```diff
@@ -5,98 +5,165 @@ nextflow.enable.dsl=2
+
+ Documentation	@ https://github.com/Sydney-Informatics-Hub/Somatic-shortV-nf
+
+Cite					@ 10.48546/workflowhub.workflow.431.1 ???
```
That is the DOI from GSV. To create a DOI for this workflow please revise the material from the workflowhub workshop you attended earlier this year.

nandan75 replied:
I will go through the workshop at https://www.biocommons.org.au/events/findable-workflows and get on it

nandan75 replied:
Will create a DOI , discuss a few things in 1-on-1

georgiesamaha commented on main.nf:
```diff
@@ -5,98 +5,165 @@ nextflow.enable.dsl=2
 def helpMessage() {
     log.info"""
-  Usage:  nextflow run main.nf --input <samples.tsv> 
+  Usage:  nextflow run main.nf --ref reference.fasta
```
Why is `--input` excluded here?

nandan75 replied:
Included now

georgiesamaha commented on main.nf:
```diff
@@ -5,98 +5,165 @@ nextflow.enable.dsl=2
-	
-""".stripIndent()
+        --whoami                    HPC user name (Setonix or Gadi HPC)
+        --gadi_account              Project accounting code for NCI Gadi (e.g. aa00)
```
In the config you've applied/defined `--gadi-account`, but here you've got `--gadi_account`

georgiesamaha commented on main.nf:
```diff
@@ -5,98 +5,165 @@ nextflow.enable.dsl=2
-} else {
+  // Split cohort file to collect info for each sample
+	bam_pair_ch = checkInputs.out
+		.splitCsv(header: true, sep:"\t")
```
Suggest changing sep to se:"," when changing input format to csv

nandan75 replied:
Changed to inout as .csv and tested

georgiesamaha commented on main.nf:
```diff
@@ -5,98 +5,165 @@ nextflow.enable.dsl=2
+  // Split cohort file to collect info for each sample
+	bam_pair_ch = checkInputs.out
+		.splitCsv(header: true, sep:"\t")
+		.map { row -> tuple(row.sampleID, file(row.bam_N), file(row.bam_T))}
```
Should we be using a method like [this](https://training.nextflow.io/advanced/grouping/) to map metadata instead of current method?

georgiesamaha commented on main.nf:
```diff
@@ -5,98 +5,165 @@ nextflow.enable.dsl=2
-	processOne(input)
+
+//params.bams = "/scratch/er01/ndes8648/pipeline_work/nextflow/INFRA-83-Somatic-ShortV/Fastq-to-BAM_BASEDIR/Fastq-to-BAM/Dedup_sort/*-{N,T}.coordSorted.dedup.bam"
+//bam_pair_ch=Channel.fromFilePairs( params.bams )
```
Remove if not using

nandan75 replied:
Done

georgiesamaha commented on modules/CalculateContamination.nf:
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl=2
```
No need for this with current version of nextflow

georgiesamaha commented on modules/CalculateContamination.nf:
```diff
@@ -0,0 +1,35 @@
+                path ("${bam_id}-T_segments.table")
+
+        shell:
+        // Calculate the fraction of reads coming from cross-sample contamination
+        
+        '''
+        
+        gatk  --java-options "-Xmx16g" \
+                CalculateContamination \
+                -I !{bam_id}-T.pileups.table \
+                -tumor-segmentation !{bam_id}-T_segments.table \
+                -matched !{bam_id}-N.pileups.table \
+                -O !{bam_id}-T_!{bam_id}-N_contamination.table
+        '''
```
How come you're using shell with ''' and not script with """?

nandan75 replied:
Since we are using the  **--java-options "-Xmx16g"** inside the bash block, 

When using  **"** inside the bash script block, we need to use 

- shell with ''' and not script with """ 
  - To recognise the **"-Xmx16g"**

georgiesamaha commented on modules/FilterMutectCalls.nf:
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env nextflow
+
+
+nextflow.enable.dsl=2
```
No need for this with current version of nextflow

nandan75 replied:
Removed

georgiesamaha commented on modules/FilterMutectCalls.nf:
```diff
@@ -0,0 +1,50 @@
+        output :
+                path ("${bam_id}-T_${bam_id}-N.filtered.vcf.gz")
+
+        shell:
```
Same comments as previous process

nandan75 replied:
Since we are using the --java-options "-Xmx16g" inside the bash block,

When using " inside the bash script block, we need to use

shell with ''' and not script with """
To recognise the "-Xmx16g"

georgiesamaha commented on modules/CalculateContamination.nf:
```diff
@@ -0,0 +1,35 @@
+
+        output:
+                path ("${bam_id}-T_${bam_id}-N_contamination.table")
+                path ("${bam_id}-T_segments.table")
```
Use the emit function to label outputs as input channels for downstream processes

nandan75 replied:
- Please test the pipeline without "emit". 
- I will push the changes with emit today.

georgiesamaha commented on modules/changed_code.nf:
What is this for? If not in use or for your private development notes, no need to push to repo

nandan75 replied:
Removed

georgiesamaha commented on modules/check_cohort.nf:
We can add functionality here I think to check integrity of files. Lets chat about this later. I started trying to do this but hit an error last week.

nandan75 replied:
Sure. Thanks

georgiesamaha reviewed (changes requested):
Not sure why the gadi profile wasn't being correctly applied to the final process. When running the PBS on the login node (i.e. with bash, not with qsub), I get the following error:

```
Error executing process > 'getFilteredVariants (getFilteredVariants Patient2)'

Caused by:
  Process requirement exceed available memory -- req: 20 GB; avail: 4 GB
```

**ND**

- I have tested this a few times now and confirmed the config files. All runs well for me and using the normalq with the assigned resources!
-  Ching-Yu has run it as well on his side. 
- **It is possibly a transient issue** - I am facing problem with gadi today when it goes very slow at times - qstat is not working etc ..
- So once I push all the changes , can you please re-run the analysis to check.

Commit:
Major README.md updates as per GS feeedback

Commit:
Changed image name and restored the new image

Commit:
A few other respnses to GS review

Commit:
Removed module load , adhoc scatter files, reverted to g/data scatter files

Commit:
tested using csv instead of tsv

Commit:
Readme.md small text update

Commit:
Tested-OK pipeline: fixed intervals and without emit

georgiesamaha commented:
What do you need me to review? How? Anything that is not functioning?

nandan75 commented:
- Please use the script PBS_gadi_runPipeline.sh
   <https://github.com/Sydney-Informatics-Hub/Somatic-shortV-nf/blob/review_testRun_from_Georgie/PBS_gadi_runPipeline.sh>
for
   testing since the last step of the pipeline did not work for you previously.
   - I will push the changes including the *emit* later.So please ignore if
   the syntax looks not standardised using emit, as suggested.
   - I am working on the 'creating intervals' - So for now please use the
   fixed intervals as in the PBS_gadi_runPipeline.sh
   <https://github.com/Sydney-Informatics-Hub/Somatic-shortV-nf/blob/review_testRun_from_Georgie/PBS_gadi_runPipeline.sh>
   - I have incorporated most of the changes as suggested by you in the
   previous review and added a few questions.

Thanks

Nandan

On Tue, 19 Dec 2023, 8:58 am Georgie Samaha, ***@***.***>
wrote:

> What do you need me to review? How? Anything that is not functioning?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Sydney-Informatics-Hub/Somatic-shortV-nf/pull/2#issuecomment-1861756445>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC4A5CNBJOOYXKI232ZORPLYKC4BLAVCNFSM6AAAAABASQ7FDWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNRRG42TMNBUGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

georgiesamaha commented on PBS_gadi_runPipeline.sh:
Formatting of this script is very messy. Please remove multiple new lines between sections, group variables together. You could look into using a linting tool like editorConfig and/or Prettier to automatically format your code.

georgiesamaha commented on PBS_gadi_runPipeline.sh:
```diff
@@ -0,0 +1,51 @@
+  
+
+
+# Run the pipeline (remove annotsv if not needed)
```
This has been copied directly from the GSV pipeline, please fix.

georgiesamaha replied:
This is still not resolved.

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,226 @@
+The pipeline is written in Nextflow and uses Singularity/Docker to run containerised tools.
+
+There are two main steps to this workflow 
+- First step is to generate a large set of candidate somatic variants using the tool  [Mutect2](https://gatk.broadinstitute.org/hc/en-us/articles/360035531132). 
+- The next step is to filter the candidate variants to obtain a more confident set of somatic variant calls. 
+- The third (optional) step is to annotate the variants with gene level information using an external genomic variant annotations and functional effect prediction tool called [SnpEff](http://pcingola.github.io/SnpEff/).
```
The wording here is too verbose. Instead of dot points just number them 1-2.

nandan75 replied:
- I have changed this to numbers 
- The snpEff tool involves additional requirements such as setting up the database etc (by the user). I will add the tool in the next update.

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,226 @@
+## Diagram
+
+<p align="center"> 
+<img src="./images/Somatic_variant_calling.png" width="60%">
```
This renders too small, the text is not readable.

nandan75 replied:
Increased the size

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,226 @@
+</p> 
+
 
 ## Workflow description 
```
Remove this section if not in use

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,226 @@
+
+To run this pipeline you will need the following reference files:
+
+* Indexed reference genome in FASTA format 
```
Clarify what indexes are required (samtools, bwa, gatk?)

nandan75 replied:
Please refer to https://github.com/Sydney-Informatics-Hub/Somatic-shortV-nf/pull/2#discussion_r1432015491 below

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,226 @@
+
+* Indexed reference genome in FASTA format 
+
+You will need to download and index a copy of the reference genome you would like to use. Reference FASTA files must be accompanied by a .fai index file. If you are working with a species that has a public reference genome, you can download FASTA files from the [Ensembl](https://asia.ensembl.org/info/data/ftp/index.html), [UCSC](https://genome.ucsc.edu/goldenPath/help/ftp.html), or [NCBI](https:/ ...
```
Was this also taken directly from GSV? Does this actually apply to the Somatic pipeline or are there other important considerations? Please don't copy stuff over from other workflows without checking.

nandan75 replied:
Checked now.

GATK (the only tool) requires an indexed reference fasta as input. So the above statement is valid here.

georgiesamaha replied:
GATK typically requires specific indexes. The ref indexes currently on gdata that you're using comprise bwa, samtools, and gatk. Ensure you know exactly which indexes need to be produced.

nandan75 replied:
thanks.

- **samtools**      .fai (for indexing FASTA files)
- **bwa**.    .amb, .ann, .bwt, .pac, .sa (for BWA index files)
- **picard** : .dict (for sequence dictionaries)

I will include these details about what and how they can be generated

georgiesamaha replied:
Are all of these ae actually needed to execute the processes though

nandan75 replied:
- We require  **.fai** 
- We start with bam - So bwa is used by [Fastq-to-BAM](https://github.com/Sydney-Informatics-Hub/Fastq-to-BAM) which we recommend in the Readme.md. So we may not want to mention it in this Readme...
-  **.dict** is used for estimating intervals in generating intervals files based on "input interval number" as shown [here](https://github.com/Sydney-Informatics-Hub/Fastq-to-BAM/blob/fastq-to-bam-v2/create_gatk_ref_intervals.pbs). However if we go by using the "input interval number" to get exact intervals, we do not require it. I will confirm.

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,226 @@
+├── scripts/
+├── main.nf
+├── modules/
+├── PBS_gadi_runPipeline.sh
```
This needs to move into the `scripts/` directory.

nandan75 replied:
Yes - moved.

georgiesamaha commented on README.md:
````diff
@@ -1,15 +1,226 @@
+
+### 4. Run the pipeline 
+
+The most basic run command for this pipeline is: 
+
+```
+nextflow run main.nf --input  sample.csv --ref reference.fasta
+
+```
````
Again, is this copied directly from GSV? It is currently missing key elements.

nandan75 replied:
Changed to this pipeline default command
nextflow run main.nf --input samples.csv --ref reference.fasta  --intervalList_path path_to_intervals --ponvcf pon.vcf.gz

For some reason I am still struggling at the last step of - dynamic interval generation
What works
- Creating the intervals based on a integer input  "number of intervals" using https://github.com/Sydney-Informatics-Hub/Fastq-to-BAM/blob/fastq-to-bam-v2/create_gatk_ref_intervals.pbs 
- Iterating over each interval to start a Mutect2 scatter comamnd
- What does not work (yet)
   - Mutect2 command
  - On creating the GATK intervals based on "number of intervals" value is throwing a compatibility issue. Obviously I am missing something very small but cruciual
  - I am trying to get my head around it - I will update and test the pipeline once this works. Till then the fixed intervals (precomputed and placed in a local path) are being used 

Once this works
The hardcore path --intervalList_path will be removed, interval scatter files will be generated and will be used without defining the 'intervalList' in nextflow.config. Almost there!!!!

georgiesamaha commented on README.md:
Have only partially reviewed the README as it is currently lacking specific instructions for SomaticShortV, contains a lot of inaccurate details. Please do not copy directly from other workflow repositories without checking.

georgiesamaha commented on images/Somatic_variant_calling.png:
Image rendered in README is too small. Cannot read text.

nandan75 replied:
Increased size

georgiesamaha commented on main.nf:
```diff
@@ -1,102 +1,163 @@
+include{  CalculateContamination                     } from './modules/CalculateContamination'
+include { FilterMutectCalls                          } from './modules/FilterMutectCalls'
+include { getFilteredVariants                        } from './modules/getFilteredVariants'
+//include { annotateWithSnpEff                        } from './modules/annotateWithSnpEff'
```
Is this in use or still to be added? When running the workflow, I got the following message: 
```
WARN: There's no process matching config selector: annotateWithSnpEff 
```

nandan75 replied:
I am removing snpEff related process for now. There are some additional requirements with regards to - downloading the snpEff database etc (by the user). Need clarity about whats the best approach. I will finalsie this in next update

georgiesamaha commented on main.nf:
```diff
@@ -1,102 +1,163 @@
 Workflow run parameters 
 =======================================================================================
+
 input       : ${params.input}
 outDir      : ${params.outDir}
 workDir     : ${workflow.workDir}
```
This does not yet capture all parameters applied in this workflow.

georgiesamaha commented on main.nf:
Same comment as for README, the main.nf while functional is still incomplete (see lines: 71-91, 100) and appears to have been copied directly from GSV without checking or adapting for the needs of this pipeline. I will review once these issues have been addressed.

georgiesamaha commented on modules/annotateWithSnpEff.nf:
Assume this is still a work in progress, have not reviewed.

nandan75 replied:
Please ignore snpEff for this version. I have tried to remove all references to this last prcocess.
I have to figure out a few things around the user downloading the snpEff db.

georgiesamaha commented on nextflow.config:
```diff
@@ -1,85 +1,148 @@
+    }
+
+// Job profiles
+profiles {
```
Remove profiles that are not currently tested and functional

nandan75 replied:
ok. Removed setonix and nimbus profiles from - 

Should we keep the setonix.config and nimbus.config template files or delete?

georgiesamaha replied:
Remove any config that is not yet functional

georgiesamaha commented on samples.tsv:
Add to gitignore

georgiesamaha commented on samples.csv:
Add to gitignore

georgiesamaha reviewed (commented):
Workflow executed without error. Haven't been able to complete review as `README` and `main.nf` are incomplete and lacking essential information re: paramters and reference data. 

Additionally, when everything is functional you'll need to tidy the code base to ensure formatting is consistent and readable. I suggest after a manual tidy that you implement editorconfig and prettier linting tools to automate formatting.

nandan75 commented:
Hi @georgiesamaha 

- I have updated the main.nf and Readme.md files.
- The snpEff process is removed for now. I will add it in the next immediate update
- The auto generation of intervals based on "number of intervals" input value is giving some compatibility issues (all done but the format of the scatter files seems to be unhappy). I am checking and rechecking ...But still not there..
-  **NCI-GADI  has gone real slow** - qstat command not working for now.
 -

georgiesamaha commented:
Same as last time, please let me know what/how you want me to review. And request my reivew on this PR on the side menu

Commit:
Changes main.nf, Readme.md

nandan75 commented:
Please have a look now.


   - I have updated the main.nf and Readme.md files.
   - The snpEff process is removed for now. I will add it in the next
   immediate update
   - The auto generation of intervals based on "number of intervals" input
   value is giving some compatibility issues (all done but the format of the
   scatter files seems to be unhappy). I am checking and rechecking ...But
   still not there..
   - *NCI-GADI has gone real slow* - qstat command not working for now.




On Wed, Dec 20, 2023 at 2:47 PM Georgie Samaha ***@***.***>
wrote:

> Same as last time, please let me know what/how you want me to review
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Sydney-Informatics-Hub/Somatic-shortV-nf/pull/2#issuecomment-1863803269>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC4A5CP654YO7QGSHZD7ED3YKJNUXAVCNFSM6AAAAABASQ7FDWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNRTHAYDGMRWHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

georgiesamaha commented:
It is not clear to me what I am reviewing. How am I expected to run the workflow, what data am I using? If this is the final test before pushing to main, then I expect testing will need to be more extensive than above in this PR.

nandan75 commented:
Please have a look at the
- main.nf file
- Readme.md

- Since I am encountering issues to run the command on gadi - Please avoid
running the actual command for sometime
   - I will confirm  that the version of the code in the github repo works
before I merge.
   -  The auto generation of intervals based on "number of intervals" input
value is giving some compatibility issues (all done but the format of the
scatter files seems to be unhappy).  So the most updated version of the
script is similar to the one you tested.






On Wed, Dec 20, 2023 at 2:55 PM Georgie Samaha ***@***.***>
wrote:

> It is not clear to me what I am reviewing. How am I expected to run the
> workflow, what data am I using? If this is the final test before pushing to
> main, then I expect testing will need to be more extensive than above in
> this PR.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Sydney-Informatics-Hub/Somatic-shortV-nf/pull/2#issuecomment-1863807718>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC4A5CLC54L3AEGOL6X7OFDYKJOSVAVCNFSM6AAAAABASQ7FDWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNRTHAYDONZRHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

georgiesamaha commented on PBS_gadi_runPipeline.sh:
```diff
@@ -0,0 +1,43 @@
+module load nextflow/22.04.3
+module load singularity
+
+
```
Needs a header explaining what this does

georgiesamaha commented on PBS_gadi_runPipeline.sh:
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+  
+#PBS -P er01 
```
Remove project code

georgiesamaha commented on PBS_gadi_runPipeline.sh:
```diff
@@ -0,0 +1,43 @@
+#PBS -N Somatic-shortV
+#PBS -l walltime=02:00:00
+#PBS -l ncpus=1
+#PBS -l mem=60GB
```
Is 60gb required to run the head job? I suspect not

nandan75 replied:
- The gatk container is a big one. It did crash once due to memory related issues. So changed this value to a bit bigger
- I will test once more

georgiesamaha commented on PBS_gadi_runPipeline.sh:
```diff
@@ -0,0 +1,43 @@
+#PBS -e Somatic-shortV-nf.e
+#PBS -o Somatic-shortV-nf.o
+#PBS -l wd
+#PBS -l storage=scratch/er01+gdata/er01
```
Remove project code

georgiesamaha commented on PBS_gadi_runPipeline.sh:
```diff
@@ -0,0 +1,43 @@
+
+# Fill in these variables for your run
+
+samples=/scratch/er01/ndes8648/pipeline_work/nextflow/INFRA-83-Somatic-ShortV/Somatic-shortV-nf/samples.csv
+ponvcf=/scratch/er01/ndes8648/pipeline_work/nextflow/INFRA-83-Somatic-ShortV/Somatic-shortV-nf/pon.vcf.gz
+ref=/g/data/er01/SIH-HPC-WGS/Reference/hs38DH.fasta
+small_exac_common=/g/data/er01/SIH-HPC-WGS/Reference/gatk-best-practices/somatic-hg38/small_exac_common_3.hg38.vcf.gz
```
None of these should be hardcoded. Users should fill in for themselves.

georgiesamaha commented on PBS_gadi_runPipeline.sh:
```diff
@@ -0,0 +1,43 @@
+intervalList_path=/g/data/er01/SIH-HPC-WGS/Reference/BQSR_apply_intervals_13
+
+outDir=results
+whoami=npd561  
```
This does not need to be defined, `$(whoami)` is a bash command that picks up username

georgiesamaha commented on PBS_gadi_runPipeline.sh:
```diff
@@ -0,0 +1,43 @@
+nextflow run main.nf -resume \
+        --input ${samples} \
+        -profile gadi \
+        --whoami ${whoami} --gadi_account $PROJECT \
```
change `${whoami}` to `$(whoami)`

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,228 @@
+The pipeline is written in Nextflow and uses Singularity/Docker to run containerised tools.
+
+There are two main steps to this workflow 
+1. First step is to generate a large set of candidate somatic variants using the tool  [Mutect2](https://gatk.broadinstitute.org/hc/en-us/articles/360035531132). 
```
Reword as: 

```
1. Generate a large set of candidate somatic variants using the tool [Mutect2](https://gatk.broadinstitute.org/hc/en-us/articles/360035531132)
```

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,228 @@
+
+There are two main steps to this workflow 
+1. First step is to generate a large set of candidate somatic variants using the tool  [Mutect2](https://gatk.broadinstitute.org/hc/en-us/articles/360035531132). 
+2. The next step is to filter the candidate variants to obtain a more confident set of somatic variant calls. 
```
Reword as: 
```
2. Filter the candidate variants to obtain a more confident set of somatic variant calls. 
```

georgiesamaha commented on README.md:
````diff
@@ -1,15 +1,228 @@
+```csv
+sampleID,bam-N,bam-T 
+SAMPLE1,/data/Bams/sample1-N.bam,/data/Bams/sample1-T.bam
+SAMPLE2,/data/Bams/sample2-N.bam,/data/Bams/sample2-T.bam|
````
Typo at end of second sample line

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,228 @@
+
+### 2. Prepare the reference materials 
+
+To run this pipeline you will need the following reference files:
+
+- Indexed reference genome in FASTA format. You will need to download and index a copy of the reference genome you would like to use. Reference FASTA files must be accompanied by a .fai index file. If you are working with a species that has a public reference genome, you can download FASTA files from the [Ensembl](https://asia.ensembl.org/info/data/ftp/index.html), [UCSC](https://genome.ucsc.edu/ ...
+- You can use our [IndexReferenceFasta-nf pipeline](https://github.com/Sydney-Informatics-Hub/IndexReferenceFasta-nf) to generate indexes. This pipeline uses the following tools for generating specific index files.
```
This is very hard to read and is not gramatically correct. 

Please change to list out the types of indexes required. Remove the box below, instead place each in a separate dot point with an embedded link to the samtools index documentation, gatk index documentation, bwa index documentation. Like: 

```
* [Samtools](link) .fai
``` 

Below that place the information currently included as dot points into a small paragraph without using the dot point.

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,228 @@
+- Indexed reference genome in FASTA format. You will need to download and index a copy of the reference genome you would like to use. Reference FASTA files must be accompanied by a .fai index file. If you are working with a species that has a public reference genome, you can download FASTA files from the [Ensembl](https://asia.ensembl.org/info/data/ftp/index.html), [UCSC](https://genome.ucsc.edu/ ...
+- You can use our [IndexReferenceFasta-nf pipeline](https://github.com/Sydney-Informatics-Hub/IndexReferenceFasta-nf) to generate indexes. This pipeline uses the following tools for generating specific index files.
+
+|Tool    | Index file(s)   |
+|-------------------|:--------------------------------- |
+|samtools            | .fai                               |
+|bwa            | .amb, .ann, .bwt, .pac, .sa                               |
+|picard            | .dict                              |
+
```
Remove as per comment above

georgiesamaha commented on README.md:
````diff
@@ -1,15 +1,228 @@
+```
+***Note***: You must specify the full path for the reference fasta, even if it is in your working directory.
+
+- You will also need to download 
````
???

georgiesamaha commented on README.md:
````diff
@@ -1,15 +1,228 @@
+* **modules** contains individual process files for each step in the workflow. 
+* **config** contains infrastructure-specific config files (this is currently under development)
+
+### 4. Run the pipeline 
+
+The most basic run command for this pipeline is: 
+
+```
+nextflow run main.nf --input samples.csv --ref reference.fasta  --intervalList_path path_to_intervals --ponvcf pon.vcf.gz
+        
+```
+
+By default, this will generate `work` directory, `results` output directory and a `runInfo` run metrics directory inside the results directory. 
+
+To specify additional optional tool-specific parameters, see what flags are supported by running:
+
+```
+nextflow run main.nf --help 
+```
+
+If for any reason your workflow fails, you are able to resume the workflow from the last successful process with `-resume`. 
````
Is this currently working? If not, then you need to clarify that it is currently not functional.

nandan75 replied:
I have tested the pipeline. It works fine.

The description in Readme.md should assist the user. Please give it a try.

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,228 @@
+
+
+### 5. Results 
+Once the pipeline is complete, you will find all outputs for each sample in the results directory. **Note** I still need to delete arrange the output files in proper sub-directory structure.
```
Remove your personal note from the README.

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,228 @@
+
+## Infrastructure usage and recommendations
+
+This pipeline has been successfully implemented on [NCI Gadi](https://nci.org.au/our-systems/hpc-systems), [Pawsey Setonix HPC](https://pawsey.org.au/systems/setonix/) and [Pawsey Nimbus cloud](https://pawsey.org.au/systems/nimbus-cloud-service/) using infrastructure-specific configs. These configs can be used to interact with the job scheduler and assign a project code to all task job submission ...
```
This isn't currently true. It is currently only functioning on Gadi

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,228 @@
+This pipeline has been successfully implemented on [NCI Gadi](https://nci.org.au/our-systems/hpc-systems), [Pawsey Setonix HPC](https://pawsey.org.au/systems/setonix/) and [Pawsey Nimbus cloud](https://pawsey.org.au/systems/nimbus-cloud-service/) using infrastructure-specific configs. These configs can be used to interact with the job scheduler and assign a project code to all task job submission ...
+
+* `--whoami` your NCI or Pawsey user name
+* `--setonix_account` the Setonix project account you would like to bill service units to
```
Remove this

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,228 @@
+
+* `--whoami` your NCI or Pawsey user name
+* `--setonix_account` the Setonix project account you would like to bill service units to
+* `--gadi-account` the Gadi project account you would like to bill service units to
```
Why is there an inconsistent use of underscores and tabs in parameters? Please make consistent.

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,228 @@
+
+|metadata field     | Somatic-shortV-nf / v1.0     |
+|-------------------|:--------------------------------- |
+|Version            | 1.0                               |
```
Change to 1.0.0

georgiesamaha commented on README.md:
```diff
@@ -1,15 +1,228 @@
+|Nextflow     |>=20.07.1 |
+|Singularity  |   3.11.3       |
+|GATK  |   4.4.0.0       |
+|SnpEff       |  5.2   |
```
Currently not included so remove

georgiesamaha commented on README.md:
````diff
@@ -1,15 +1,228 @@
+- [Nextflow documentation](https://www.nextflow.io/docs/latest/index.html)
 
 ## Help / FAQ / Troubleshooting
+* It is essential that the reference genome you're using contains the same chromosomes, contigs, and scaffolds as the BAM files. To confirm what contigs are included in your indexed BAM file, you can use Samtools idxstats:
+```
+samtools idxstats input.bam | cut -f 1
+```
````
This is taken directly from GSV. Does this even apply to this workflow?

nandan75 replied:
- The input to this pipeline are bam files (generated by other tools)
- The user will be choosing a Reference genome for this workflow.
- I thought it was a good idea for them to confirm that the bams are compatible with the reference-fasta 
  - Let me know if it makes sense or I will remove.

georgiesamaha replied:
See #9

georgiesamaha commented on main.nf:
```diff
@@ -1,102 +1,138 @@
+	CalculateContamination(bam_pair_ch,GetPileupSummaries_T.out.collect(),GetPileupSummaries_N.out.collect())
+
+  // Filter somatic SNVs and indels called by Mutect2
+	FilterMutectCalls(bam_pair_ch,MergeMutectStats.out[1].collect(),CalculateContamination.out[0].collect(),CalculateContamination.out[1].collect(),GatherVcfs.out[0].collect(),GatherVcfs.out[1].collect(),LearnReadOrientationModel.out.collect(),params.ref)	
```
Can you split this across multiple lines so it reads cleaner?

georgiesamaha commented on samples.csv:
Remove this or gitignore

georgiesamaha commented on samples.tsv:
Remove this or gitignore

georgiesamaha commented on scripts/pullContainers.sh:
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+#PBS -P er01
```
Remove project code

georgiesamaha commented on scripts/pullContainers.sh:
```diff
@@ -0,0 +1,21 @@
+#PBS -W umask=022
+#PBS -q copyq
+#PBS -l wd
+#PBS -l storage=scratch/er01+gdata/er01
```
Remove project code

georgiesamaha commented on scripts/pullContainers.sh:
```diff
@@ -0,0 +1,21 @@
+# TODO create lint test to check this script and nextflow.config have same containers
+
+# specify singularity cache dir (consistent with gadi.config)
+export SINGULARITY_CACHE_DIR=/scratch/er01/ndes8648/pipeline_work/nextflow/INFRA-83-Somatic-ShortV/singularity_cache
```
This is still hardcoded.

georgiesamaha commented on nextflow.config:
```diff
@@ -1,85 +1,142 @@
+withName: 'mutect2' {
+        cpus = 1
+        memory = '20.GB'
+        container = 'quay.io/biocontainers/gatk4:4.4.0.0--py36hdfd78af_0'
```
This container is repeated for all processes. Is this necessary? Remember DRY principles to make for more concise, readable code.

nandan75 replied:
Done

georgiesamaha commented on nextflow.config:
```diff
@@ -1,85 +1,142 @@
+// Resources for each process
+// default run resource parameters
+process {
+
```
Can you add container here instead of repeating under each withName

nandan75 replied:
Done

georgiesamaha commented on config/gadi.config:
```diff
@@ -1,38 +1,104 @@
 singularity {
     enabled = true
-    cacheDir = "/scratch/${params.pbs_account}/${params.whoami}/.nextflow/singularity"
+    cacheDir = "/scratch/${params.gadi_account}/${params.whoami}/singularity"
```
Confirm `gadi_account` matches parameter description in nextflow.config and README

georgiesamaha reviewed (changes requested):
A few things left to tidy. Have suggested some changes, pointed out where things aren't clear to me. Please clarify in the README what is functional and what is not and ensure you're not revealing private information by pushing your account details to github

Commit:
Deidentified details from PBS_gadi_runPipeline.sh

nandan75 commented:
**Whats works**
- Pipeline works using the script `scripts/run_pipeline_on_gadi_script.sh`
 - Using the intervals downloaded from gatk website, detailed instructions as to how to go about downloading the intervals is provided in Readme.md
 - The Redme.md also provides instructions to use the bash [Somatic-ShortV](https://github.com/Sydney-Informatics-Hub/Somatic-ShortV) pipeline to generate accessory files.

**What does not work**
- Auto-generation of intervals based on "no_of_intervals" variable . Some issue with compatibility of the generated intervals!
- snpEff process has not been integrated - but it is almost ready to be tested.  

**What is not done**
- Full dataset run with productionized version
- Benchmarking


**Please use the above script and the following variables for testing**

**samples**=/scratch/er01/ndes8648/pipeline_work/nextflow/INFRA-83-Somatic-ShortV/test_files_for_Georgie/samples.csv
**ponvcf**=/scratch/er01/ndes8648/pipeline_work/nextflow/INFRA-83-Somatic-ShortV/test_files_for_Georgie/pon.vcf.gz
**ref**=/g/data/er01/SIH-HPC-WGS/Reference/hs38DH.fasta
**small_exac_common**=/g/data/er01/SIH-HPC-WGS/Reference/gatk-best-practices/somatic-hg38/small_exac_common_3.hg38.vcf.gz
**intervalList_path**=/scratch/er01/ndes8648/pipeline_work/nextflow/INFRA-83-Somatic-ShortV/test_files_for_Georgie/interval_files

Commit:
Tested,Updated as on Dec 20th 6.30pm

georgiesamaha commented:
I will push current changes to main and include note that the workflow is currently not functional on README. Required upcoming work to wrap this up has been documented as issues #4 , #5, and #6.

georgiesamaha committed:
Add not functional warning to README

georgiesamaha committed:
Clean variables from gadi script

georgiesamaha committed:
Clean morevariables from gadi script

georgiesamaha committed:
Remove resume flag from gadi execution command

georgiesamaha committed:
Clean README description

georgiesamaha committed:
Fix typo in README

georgiesamaha committed:
Clean ignore

georgiesamaha committed:
Update README

georgiesamaha commented:
Have captured some TODOs in issues.
